src/common/utils/ldap/ldap.go

```diff
@@ -220,6 +220,22 @@ func (session *Session) SearchUser(username string) ([]models.LdapUser, error) {
 			}
 			u.GroupDNList = groupDNList
 		}
+
+		log.Debugf("Searching for nested groups")
+		nestedGroupDNList := []string{}
+		nestedGroupFilter := createNestedGroupFilter(ldapEntry.DN)
+		result, err := session.SearchLdap(nestedGroupFilter)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, groupEntry := range result.Entries {
+			nestedGroupDNList = append(nestedGroupDNList, strings.TrimSpace(groupEntry.DN))
+			log.Debugf("Found group %v", groupEntry.DN)
+		}
+		u.GroupDNList = append(u.GroupDNList, nestedGroupDNList...)
```

Comment:
Top level groups present in both u.GroupDNList and nestedGroupDNList, could you please remove duplicated items in the u.GroupDNList?
for example:
CN=cna_users,CN=Users,DC=harbor,DC=com is a top level group.
the u.GroupDNList might contains two items with DN: CN=cna_users,CN=Users,DC=harbor,DC=com
```
CN=cna_users,CN=Users,DC=harbor,DC=com 
CN=harbor_users,CN=Users,DC=harbor,DC=com 
CN=cna_users,CN=Users,DC=harbor,DC=com
```

Reply:
Oh right! Thanks! I'm at an appointment right now but will do when i'm back.